dynawo/sources/Models/Modelica/Dynawo/Electrical/Loads/LoadZIP.mo

```diff
@@ -0,0 +1,44 @@
+within Dynawo.Electrical.Loads;
+
+/*
+* Copyright (c) 2015-2020, RTE (http://www.rte-france.com)
+* See AUTHORS.txt
+* All rights reserved.
+* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, you can obtain one at http://mozilla.org/MPL/2.0/.
+* SPDX-License-Identifier: MPL-2.0
+*
+* This file is part of Dynawo, an hybrid C++/Modelica open source suite of simulation tools for power systems.
+*/
+
+model LoadZIP "ZIP coefficients load model"
+  extends BaseClasses.BaseLoad;
+  extends AdditionalIcons.Load;
+
+public
+  // Active power coefficents
+  parameter Real Zp "Impedance coefficient for active power";
+  parameter Real Ip "Current coefficient for active power";
+  parameter Real Pp "Power coefficient for active power";
+
+  // Reactive power coefficients
+  parameter Real Zq "Impedance coefficient for reactive power";
+  parameter Real Iq "Current coefficient for reactive power";
+  parameter Real Pq "Power coefficient for reactive power";
+
+  // In order to change the load set-point, connect an event to PRefPu or QRefPu
+  Connectors.ImPin PRefPu (value (start = s0Pu.re)) "Active power request";
+  Connectors.ImPin QRefPu (value (start = s0Pu.im)) "Reactive power request";
+
+equation
+  if (running.value) then
+    PPu = PRefPu.value * (Zp * (ComplexMath.'abs' (terminal.V) / ComplexMath.'abs' (u0Pu)) ^ 2 + Ip * (ComplexMath.'abs' (terminal.V) / ComplexMath.'abs' (u0Pu)) + Pp);
+    QPu = QRefPu.value * (Zq * (ComplexMath.'abs' (terminal.V) / ComplexMath.'abs' (u0Pu)) ^ 2 + Iq * (ComplexMath.'abs' (terminal.V) / ComplexMath.'abs' (u0Pu)) + Pq);
+  else
+    terminal.i = Complex(0);
+  end if;
+
+annotation(preferredView = "text",
+    Documentation(info = "<html><head></head><body>The ZIP coefficient load model, also known as the polynomial model, represents the variation with voltage of a load composed of a constant impedance part Z, a constant current part I and a constant power part P.&nbsp;<div><br></div><div>The coefficients Zp, Ip and Pp are the ZIP coefficients for active power and the coefficients Zq, Iq and Pq the ZIP coefficients for reactive power.&nbsp;</div><div><br></div><div>By taking Zp = Ip = Zq = Iq = 0 and Pp = Pq = 1, one will have a constant PQ load: P = PRefPu and Q = QRefPu.</div><div>By taking Zp = Pp = Zq = Pq = 0 and Ip = Iq = 1, one will get a constant current load: P = PRefPu * (U/U0) and Q = QRefPu * (U/U0).</div><div>By taking Ip = Pp = Iq = Pq = 0 and Zp = Zq = 1, one will end up with a constant impedance load: P = PRefPu * (U/U0)² and Q = QRefPu * (U/U0)²&nbsp;</div></body></html>"));
```

Comment:
Are the coefficient always 0 or 1 or could we have a mixed model? Is the sum always equal to 1 (as if the coefficients represent the share of the load that is of type Z, I or P)? I am trying to see if we could be more specific than "parameter Real" in the coefficient definition

Reply:
You can have mixed models, depending on the equipment you try to model (very various values, either positive or negative as can be seen here for example http://engineering.nyu.edu/power/sites/engineering.nyu.edu.power/files/uploads/Experimental%20Determination%20of%20the%20ZIP%20Coefficients%20for%20Modern%20Residential,%20Commercial%20and%20Industrial%20Loads.pdf).

Nevertheless, the sum is always 1 to ensure P = P0 when U = U0

Reply:
Ok thanks